### PR TITLE
Add CCMS Allpay integration

### PR DESCRIPTION
**Pull request by adrianclay**
## What does this pull request do?

Partially addresses #21

## What is the intent behind these changes?

Generated images:
image

image

### CODE REVIEW

File: src/main/kotlin/model/Allpay.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.structurizr.view.ViewSet
+
+class Allpay private constructor() {
+  companion object : LAASoftwareSystem {
+    lateinit var system: SoftwareSystem
+
+    override fun defineModelEntities(model: Model) {
+      system = model.addSoftwareSystem(
+        "allpay",
+        "Direct Debit payment procesor"
```
Comment (heidar) [inline src/main/kotlin/model/Allpay.kt:14]: Typo
```suggestion
        "Direct Debit payment processor"
```

File: src/main/kotlin/model/CCMS.kt [hunk 2]
```diff
@@ -133,6 +133,18 @@ class CCMS private constructor() {
         "Pulls debt acknowledgements, payments collected, and debts closed. (AKA MOD502)",
         "FTP"
       )
+
+      soa.uses(
+        Allpay.system,
+        "Pushes Direct Debit mandate instructions (AKA MOD333)",
+        "FTP"
```
Comment (heidar) [inline src/main/kotlin/model/CCMS.kt:140]: N I C E

**Review by heidar — CHANGES_REQUESTED** [review]
Just a typo, otherwise shippable!

File: src/main/kotlin/model/CCMS.kt [hunk 3]
```diff
@@ -133,6 +133,18 @@ class CCMS private constructor() {
...
+      )
+
```
[the remark this answers is not in the document]
Comment (adrianclay) [reply src/main/kotlin/model/CCMS.kt:142]: Yeah I did notice that.  I felt like the default behaviour "works", but is also fragile to that "using the first relationship" behaviour.  I'll have a think.

**Review by heidar — APPROVED** [review]

**Pull request merged**